[Bug]: Flashinfer is still JITting its kernels for blackwell in prefill

### Contact Details

_No response_

### What happened?

While testing an update to the images on blackwell I encountered this error, because the CUTLASS_MLA prefill config I was using (flashinfer, DP=16, deepep ht) didn't have all of the cutlass source modules built.

However, our build OOTB should work on blackwell so we want to ensure this is AOT compiled.

```
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] WorkerProc hit an exception.
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] Traceback (most recent call last):
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm/lib64/python3.12/site-packages/flashinfer/jit/cpp_ext.py", line 258, in run_ninja
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     subprocess.run(
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/usr/lib64/python3.12/subprocess.py", line 573, in run
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     raise CalledProcessError(retcode, process.args,
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] subprocess.CalledProcessError: Command '['ninja', '-v', '-C', '/var/cache/vllm/flashinfer/.cache/flashinfer/100_90/cached_ops', '-f', '/var/cache/vllm/flashinfer/.cache/flashinfer/100_90/cached_ops/fmha_cutlass_sm100a/build.ninja']' returned non-zero exit status 1.
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] 
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] The above exception was the direct cause of the following exception:
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] 
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] Traceback (most recent call last):
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm-source/vllm/v1/executor/multiproc_executor.py", line 666, in worker_busy_loop
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     output = func(*args, **kwargs)
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]              ^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm/lib64/python3.12/site-packages/torch/utils/_contextlib.py", line 120, in decorate_context
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     return func(*args, **kwargs)
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]            ^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm-source/vllm/v1/worker/gpu_worker.py", line 447, in execute_model
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     output = self.model_runner.execute_model(scheduler_output,
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm/lib64/python3.12/site-packages/torch/utils/_contextlib.py", line 120, in decorate_context
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     return func(*args, **kwargs)
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]            ^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm-source/vllm/v1/worker/gpu_model_runner.py", line 2257, in execute_model
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     ) = self._prepare_inputs(scheduler_output)
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm-source/vllm/v1/worker/gpu_model_runner.py", line 1244, in _prepare_inputs
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     ubatch_id=ubid).build(
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]                     ^^^^^^
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm-source/vllm/v1/attention/backends/mla/common.py", line 865, in build
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     self._build_fi_prefill_wrappers(attn_metadata.prefill)
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm-source/vllm/v1/attention/backends/mla/common.py", line 592, in _build_fi_prefill_wrappers
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     self._fi_prefill_main.plan(
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm/lib64/python3.12/site-packages/flashinfer/prefill.py", line 2690, in plan
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     self._cached_module = get_fmha_module(*new_get_module_args)
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm/lib64/python3.12/site-packages/flashinfer/prefill.py", line 89, in get_fmha_module
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     ).build_and_load()
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]       ^^^^^^^^^^^^^^^^
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm/lib64/python3.12/site-packages/flashinfer/jit/core.py", line 160, in build_and_load
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     self.build(verbose, need_lock=False)
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm/lib64/python3.12/site-packages/flashinfer/jit/core.py", line 140, in build
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     run_ninja(jit_env.FLASHINFER_JIT_DIR, self.ninja_path, verbose)
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]   File "/opt/vllm/lib64/python3.12/site-packages/flashinfer/jit/cpp_ext.py", line 270, in run_ninja
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]     raise RuntimeError(msg) from e
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] RuntimeError: Ninja build failed. Ninja output:
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] ninja: Entering directory `/var/cache/vllm/flashinfer/.cache/flashinfer/100_90/cached_ops'
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] [1/2] /usr/local/cuda/bin/nvcc --generate-dependencies-with-compile --dependency-output fmha_cutlass_sm100a/fmha_cutlass_sm100.cuda.o.d -DTORCH_EXTENSION_NAME=fmha_cutlass_sm100a -DTORCH_API_INCLUDE_EXTENSION_H -DPy_LIMITED_API=0x03090000 -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1018\" -D_GLIBCXX_USE_CXX11_ABI=1 -isystem /usr/include/python3.12 -isystem /opt/vllm/lib64/python3.12/site-packages/torch/include -isystem /opt/vllm/lib64/python3.12/site-packages/torch/include/torch/csrc/api/include -isystem /usr/local/cuda/include -isystem /usr/local/cuda/include/cccl -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/include -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/csrc -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/cutlass/include -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/cutlass/tools/util/include -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/spdlog/include --compiler-options=-fPIC --expt-relaxed-constexpr -static-global-template-stub=false -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -O3 -std=c++17 --threads=1 -use_fast_math -DFLASHINFER_ENABLE_F16 -DFLASHINFER_ENABLE_BF16 -DFLASHINFER_ENABLE_FP8_E4M3 -DFLASHINFER_ENABLE_FP8_E5M2 -DNDEBUG -gencode=arch=compute_100a,code=sm_100a -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -c /opt/vllm/lib/python3.12/site-packages/flashinfer/data/csrc/fmha_cutlass_sm100.cu -o fmha_cutlass_sm100a/fmha_cutlass_sm100.cuda.o 
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] FAILED: [code=1] fmha_cutlass_sm100a/fmha_cutlass_sm100.cuda.o 
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] /usr/local/cuda/bin/nvcc --generate-dependencies-with-compile --dependency-output fmha_cutlass_sm100a/fmha_cutlass_sm100.cuda.o.d -DTORCH_EXTENSION_NAME=fmha_cutlass_sm100a -DTORCH_API_INCLUDE_EXTENSION_H -DPy_LIMITED_API=0x03090000 -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1018\" -D_GLIBCXX_USE_CXX11_ABI=1 -isystem /usr/include/python3.12 -isystem /opt/vllm/lib64/python3.12/site-packages/torch/include -isystem /opt/vllm/lib64/python3.12/site-packages/torch/include/torch/csrc/api/include -isystem /usr/local/cuda/include -isystem /usr/local/cuda/include/cccl -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/include -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/csrc -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/cutlass/include -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/cutlass/tools/util/include -isystem /opt/vllm/lib/python3.12/site-packages/flashinfer/data/spdlog/include --compiler-options=-fPIC --expt-relaxed-constexpr -static-global-template-stub=false -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -O3 -std=c++17 --threads=1 -use_fast_math -DFLASHINFER_ENABLE_F16 -DFLASHINFER_ENABLE_BF16 -DFLASHINFER_ENABLE_FP8_E4M3 -DFLASHINFER_ENABLE_FP8_E5M2 -DNDEBUG -gencode=arch=compute_100a,code=sm_100a -DFLASHINFER_ENABLE_FP8_E8M0 -DFLASHINFER_ENABLE_FP4_E2M1 -c /opt/vllm/lib/python3.12/site-packages/flashinfer/data/csrc/fmha_cutlass_sm100.cu -o fmha_cutlass_sm100a/fmha_cutlass_sm100.cuda.o 
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] In file included from /opt/vllm/lib/python3.12/site-packages/flashinfer/data/include/flashinfer/attention/blackwell/fmha_cutlass_sm100.cuh:19,
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]                  from /opt/vllm/lib/python3.12/site-packages/flashinfer/data/csrc/fmha_cutlass_sm100.cu:16:
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671] /opt/vllm/lib/python3.12/site-packages/flashinfer/data/include/flashinfer/attention/blackwell/collective/fmha_fusion.hpp:33:10: fatal error: cute/tensor.hpp: No such file or directory
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]    33 | #include "cute/tensor.hpp"
(APIServer pid=1) (EngineCore_DP6 pid=2454) (Worker_DP6_EP6 pid=2516) ERROR 10-30 17:32:00 [v1/executor/multiproc_executor.py:671]       |          ^~~~~~~~~~~~~~~~~
```

### Version

0.0.1

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

--- a/docker/Dockerfile.aws
+++ b/docker/Dockerfile.aws
@@ -293,7 +293,11 @@ RUN --mount=type=cache,target=/root/.cache/uv \
     cd flashinfer && \
     uv pip install -e . --no-build-isolation && \
     uv build --wheel --no-build-isolation --out-dir /wheels && \
-    cd .. && rm -rf flashinfer && \
+    cd flashinfer-cubin && \
+    uv pip install -e . --no-build-isolation --out-dir /wheel && \
+    cd ../flashinfer-jit-cache && \
+    FLASHINFER_CUDA_ARCH_LIST="9.0a 10.0a" uv pip install -e . --no-build-isolation --out-dir /wheel && \
+    cd ../.. && rm -rf flashinfer && \
     \
     # Build DeepEP wheel
     git clone "${DEEPEP_URL}" deepep && \

--- a/docker/scripts/cuda/builder/build-compiled-wheels.sh
+++ b/docker/scripts/cuda/builder/build-compiled-wheels.sh
@@ -37,7 +37,11 @@ cd flashinfer
 git checkout -q "${FLASHINFER_VERSION}"
 git submodule update --init --recursive
 uv build --wheel --no-build-isolation --out-dir /wheels
-cd ..
+cd flashinfer-cubin && \
+uv build --wheel --no-build-isolation --out-dir /wheels
+cd ../flashinfer-jit-cache && \
+FLASHINFER_CUDA_ARCH_LIST="9.0a 10.0a" uv build --wheel --no-build-isolation --out-dir /wheels
+cd ../../ && \
 rm -rf flashinfer
 
 # build DeepEP wheel

